### PR TITLE
DDCE-4256 Class cast exception fix

### PR DESCRIPTION
**Pull request by gastonracui**

### CODE REVIEW

File: test/unit/controllers/CaseControllerSpec.scala [hunk 1]
```diff
@@ -73,6 +73,7 @@ class CaseControllerSpec extends ControllerBaseSpec with BeforeAndAfterEach {
     atarController,
     correspondenceController,
     miscellaneousController,
+    redirectController,
```
Comment (mywyau) [inline test/unit/controllers/CaseControllerSpec.scala:76]: small change could we name this redirectService and in other controllerspecs

File: app/service/RedirectService.scala [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import models.request.AuthenticatedCaseRequest
+import play.api.mvc.ControllerHelpers._
+import play.api.mvc.Results.Redirect
+import play.api.mvc._
+
+import javax.inject.Inject
+
+class RedirectService @Inject() () {
+
+  def redirectApplication(reference: String, fileId: Option[String] = None)(
```
Comment (mywyau) [inline app/service/RedirectService.scala:29]: I know this is somewhat tested in the other controller specs, but could we get a test spec for this covering each case?

File: test/unit/controllers/CaseControllerSpec.scala (hunk 1 above)
Comment (gastonracui) [reply test/unit/controllers/CaseControllerSpec.scala:76]: nice catch, done!

File: test/unit/controllers/v2/MiscellaneousControllerSpec.scala [hunk 3]
```diff
@@ -112,5 +113,28 @@ class MiscellaneousControllerSpec extends ControllerBaseSpec with BeforeAndAfter
...
+      when(
+        eventService
+          .getFilteredEvents(any[String], any[Pagination], any[Option[Set[EventType.Value]]])(any[HeaderCarrier])
+      ) thenReturn Future(pagedEvent)
+
+      when(fileService.initiate(any[FileStoreInitiateRequest])(any[HeaderCarrier])) thenReturn Future.successful(
+        initiateResponse
+      )
+
+      when(queueService.getAll) thenReturn Future(queues)
```
Comment (mywyau) [inline test/unit/controllers/v2/MiscellaneousControllerSpec.scala:129]: these mocks can be deleted as they are only required if the code hits the renderView case, but in this case we only want to redirect.

File: test/unit/controllers/v2/CorrespondenceControllerSpec.scala [hunk 4]
```diff
@@ -116,6 +117,28 @@ class CorrespondenceControllerSpec extends ControllerBaseSpec with BeforeAndAfte
...
+      when(
+        eventService
+          .getFilteredEvents(any[String], any[Pagination], any[Option[Set[EventType.Value]]])(any[HeaderCarrier])
+      ) thenReturn Future(pagedEvent)
+
+      when(fileService.initiate(any[FileStoreInitiateRequest])(any[HeaderCarrier])) thenReturn Future.successful(
+        initiateResponse
+      )
+
+      when(queueService.getAll) thenReturn Future(queues)
```
Comment (mywyau) [inline test/unit/controllers/v2/CorrespondenceControllerSpec.scala:133]: please delete these mocks

File: test/unit/controllers/v2/LiabilityControllerSpec.scala [hunk 5]
```diff
@@ -213,34 +218,61 @@ class LiabilityControllerSpec extends ControllerBaseSpec with BeforeAndAfterEach
...
+      when(
+        eventService
+          .getFilteredEvents(any[String], any[Pagination], any[Option[Set[EventType.Value]]])(any[HeaderCarrier])
+      ) thenReturn Future(pagedEvent)
+
+      when(fileService.initiate(any[FileStoreInitiateRequest])(any[HeaderCarrier])) thenReturn Future.successful(
+        initiateResponse
+      )
+
+      when(queueService.getAll) thenReturn Future(queues)
```
Comment (mywyau) [inline test/unit/controllers/v2/LiabilityControllerSpec.scala:246]: as in other controller specs these mocks can be deleted

File: test/unit/controllers/v2/MiscellaneousControllerSpec.scala (hunk 3 above)
Comment (mywyau) [reply test/unit/controllers/v2/MiscellaneousControllerSpec.scala:129]: ` when(fileService.getAttachments(any[Case])(any[HeaderCarrier])) thenReturn Future.successful(attachments)

      when(
        eventService
          .getFilteredEvents(any[String], any[Pagination], any[Option[Set[EventType.Value]]])(any[HeaderCarrier])
      ) thenReturn Future(pagedEvent)

      when(fileService.initiate(any[FileStoreInitiateRequest])(any[HeaderCarrier])) thenReturn Future.successful(
        initiateResponse
      )

      when(queueService.getAll) thenReturn Future(queues)`

Comment (mywyau) [reply test/unit/controllers/v2/MiscellaneousControllerSpec.scala:129]:
```
//      when(fileService.getAttachments(any[Case])(any[HeaderCarrier])) thenReturn Future.successful(attachments)

//      when(
//        eventService
//          .getFilteredEvents(any[String], any[Pagination], any[Option[Set[EventType.Value]]])(any[HeaderCarrier])
//      ) thenReturn Future(pagedEvent)
//
//      when(fileService.initiate(any[FileStoreInitiateRequest])(any[HeaderCarrier])) thenReturn Future.successful(
//        initiateResponse
//      )
//
//      when(queueService.getAll) thenReturn Future(queues)
```

**Review by mywyau — APPROVED** [review]

**Pull request merged**